[Package]: libv4l2 and libv4l1

### Package description


libv4l is a collection of libraries which adds a thin abstraction layer on
top of video4linux2 devices. The purpose of this (thin) layer is to make it
easy for application writers to support a wide variety of devices without
having to write seperate code for different devices in the same class.

### Home page URL

https://github.com/philips/libv4l

### Source code URL

https://github.com/philips/libv4l

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

Its 13 years old and giving some weird errors
```
control/libv4lcontrol.c:260:16: error: called object type 'int' is not a function or function pointer
        minor == minor(st.st_rdev))
                 ~~~~~^
control/libv4lcontrol.c:419:17: warning: implicit declaration of function 'shm_open' is invalid in C99 [-Wimplicit-function-declaration]
  if ((shm_fd = shm_open(shm_name, (O_CREAT | O_EXCL | O_RDWR),
                ^
control/libv4lcontrol.c:420:6: error: use of undeclared identifier 'S_IREAD'
                         (S_IREAD | S_IWRITE))) >= 0)
                          ^
control/libv4lcontrol.c:420:16: error: use of undeclared identifier 'S_IWRITE'
                         (S_IREAD | S_IWRITE))) >= 0)
                                    ^
control/libv4lcontrol.c:423:42: error: use of undeclared identifier 'S_IREAD'
    shm_fd = shm_open(shm_name, O_RDWR, (S_IREAD | S_IWRITE));
                                         ^
control/libv4lcontrol.c:423:52: error: use of undeclared identifier 'S_IWRITE'
    shm_fd = shm_open(shm_name, O_RDWR, (S_IREAD | S_IWRITE));
                                                   ^
1 warning and 5 errors generated.
make[1]: *** [Makefile:79: control/libv4lcontrol.o] Error 1
make[1]: Leaving directory '/data/data/com.termux/files/home/libv4l/libv4lconvert'                                          make: *** [Makefile:5: all] Error 2
```

I need it for building ssr (simple screen recorder)

--- a/packages/libv4l/build.sh
+++ b/packages/libv4l/build.sh
@@ -1,0 +1,21 @@
+TERMUX_PKG_HOMEPAGE=https://git.linuxtv.org/v4l-utils.git
+TERMUX_PKG_DESCRIPTION="Linux libraries to handle media devices"
+TERMUX_PKG_LICENSE="LGPL-2.1"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=1.22.1
+TERMUX_PKG_SRCURL=https://linuxtv.org/downloads/v4l-utils/v4l-utils-${TERMUX_PKG_VERSION}.tar.bz2
+TERMUX_PKG_SHA256=65c6fbe830a44ca105c443b027182c1b2c9053a91d1e72ad849dfab388b94e31
+TERMUX_PKG_DEPENDS="libandroid-execinfo, libandroid-glob, libjpeg-turbo"
+TERMUX_PKG_BUILD_DEPENDS="argp"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+--disable-v4l-utils
+--disable-qv4l2
+"
+
+termux_step_pre_configure() {
+	local f
+	for f in lib/*/Makefile.in; do
+		sed -i '/_la_LDFLAGS = /s/ -lpthread//' ${f}
+		sed -i '/_la_LDFLAGS = /s/ -lrt / /' ${f}
+	done
+}

--- a/packages/libv4l/lib-libv4l2-Makefile.in.patch
+++ b/packages/libv4l/lib-libv4l2-Makefile.in.patch
@@ -1,0 +1,11 @@
+--- a/lib/libv4l2/Makefile.in
++++ b/lib/libv4l2/Makefile.in
+@@ -481,7 +481,7 @@
+ libv4l2_la_SOURCES = libv4l2.c log.c libv4l2-priv.h $(am__append_1)
+ libv4l2_la_CPPFLAGS = $(CFLAG_VISIBILITY) $(ENFORCE_LIBV4L_STATIC)
+ libv4l2_la_LDFLAGS = $(LIBV4L2_VERSION) -lpthread $(DLOPEN_LIBS) $(ENFORCE_LIBV4L_STATIC)
+-libv4l2_la_LIBADD = ../libv4lconvert/libv4lconvert.la
++libv4l2_la_LIBADD = ../libv4lconvert/libv4lconvert.la -landroid-glob
+ v4l2convert_la_SOURCES = v4l2convert.c
+ v4l2convert_la_LIBADD = libv4l2.la
+ v4l2convert_la_LDFLAGS = -avoid-version -module -shared -export-dynamic

--- a/packages/libv4l/lib-libv4lconvert-control-libv4lcontrol.c.patch
+++ b/packages/libv4l/lib-libv4lconvert-control-libv4lcontrol.c.patch
@@ -1,0 +1,11 @@
+--- a/lib/libv4lconvert/control/libv4lcontrol.c
++++ b/lib/libv4lconvert/control/libv4lcontrol.c
+@@ -715,7 +715,7 @@
+ 		if (shm_name[i] == '/')
+ 			shm_name[i] = '-';
+ 
+-#ifndef ANDROID
++#ifndef __ANDROID__
+ 	/* Open the shared memory object identified by shm_name */
+ 	shm_fd = shm_open(shm_name, (O_CREAT | O_EXCL | O_RDWR), (S_IREAD | S_IWRITE));
+ 	if (shm_fd >= 0)

